packages/@lwc/synthetic-shadow/src/3rdparty/inner-text.ts

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { ArrayPush } from '@lwc/shared';
+import { innerTextGetter } from '../env/element';
+import { ELEMENT_NODE, TEXT_NODE } from '../env/node';
+import { windowGetComputedStyle, windowGetSelection } from '../env/window';
+import { childNodesGetterPatched } from '../faux-shadow/node';
+import { getOwnerWindow } from '../shared/utils';
+import { getTextContent } from './polymer/text-content';
+
+type InnerTextItem = string | number;
+
+interface SelectionState {
+    element: Element;
+    onselect: ((this: GlobalEventHandlers, ev: Event) => any) | null;
+    onselectionchange: ((this: GlobalEventHandlers, ev: Event) => any) | null;
+    onselectstart: ((this: GlobalEventHandlers, ev: Event) => any) | null;
+    ranges: Range[];
+}
+
+function getElementComputedStyle(element: Element): CSSStyleDeclaration {
+    const win = getOwnerWindow(element);
```

Comment:
I am wondering what is the motivation behind retrieving the element owner window vs. using the current window? Synthetic shadow only applies to the current window.

Reply:
`window.getSelection()` works only for the iframe window, and we rely on it in the polyfill, when calling `.innerText` on an element inside an iframe, we need to get the window for that element.

tbh, I don't know if is required for `getElementComputedStyle`, I used `getOwnerWindow` in this case to follow the same pattern used for the others.

Reply:
Sounds reasonable to me. I don't see any drawbacks with looking up the owner window on the nodes.